packages/data-stores/src/site/controls.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest, WpcomClientCredentials } from '../utils';
+import { ActionType, CreateSiteAction } from './types';
+
+export default function createControls( clientCreds: WpcomClientCredentials ) {
+	return {
+		[ ActionType.CREATE_SITE ]: async ( action: CreateSiteAction ) => {
+			const { authToken, ...providedParams } = action.params;
+			const defaultParams = {
+				blog_title: '',
+				public: -1,
+				blog_name: '',
+				client_id: clientCreds.client_id,
+				client_secret: clientCreds.client_secret,
+				lang_id: 1,
+				locale: 'en',
+				options: {
+					theme: 'pub/maywood',
+					site_segment: 1,
+					site_vertical: 'p1',
+					site_vertical_name: 'Restaurant',
+					site_information: {
+						title: 'Test Site for Gutenboarding',
+					},
+					site_creation_flow: 'gutenboarding',
+				},
+			};
+			const mergedParams = Object.assign(
+				{},
+				defaultParams,
+				{ ...providedParams },
+				{ validate: false } // Set to false because account validation should be a separate action
+			);
+			const newUser = await wpcomRequest( {
+				path: '/sites/new',
+				apiVersion: '1.1',
+				method: 'post',
+				body: mergedParams,
+				token: authToken || undefined,
```

Comment:
I'm not sure this is needed right now.  It's passed to `wpcomUndocumented` if `config.isEnabled( 'oauth' )`, but I think it's related to the OAuth login page.

See: https://github.com/Automattic/wp-calypso/blob/master/client/lib/oauth-store/README.md

Reply:
I think we definitely won't need it thanks to your reloadWpcomProxy PR in #39049! But I used `token` here just because I found the key in the `wpcomFactory` here: https://github.com/automattic/wp-calypso/blob/2de5e432fc90d3793d8bac4f8aa3e7a7dfb5432c/client/lib/wpcom-undocumented/index.js#L41 and it seemed to get it working for me in testing 🤷‍♂

But I think your idea is much better, and will allow the site creation to be properly separate from User creation (so we don't have to go passing around bearer tokens).

Reply:
> I got the wrong end of the stick

Oh, fair enough. Good to now.